File: tyama_icpc2007dA.scala
//usr/bin/env scala $0 $@;exit

object Main{
	def judge(n:Int){
		var ma=0;var mi=1000;var s=0
		for(i <- 0 until n){
			var x=Integer.parseInt(readLine())
			if(x<mi)mi=x
			if(x>ma)ma=x
			s+=x
		}
		printf("%d\n",(s-ma-mi)/(n-2))
	}

	def main(args: Array[String]){
		while(true){
			var n=Integer.parseInt(readLine())
			if(n==0)return
			judge(n)
		}
	}
}

File: tyama_icpc2007dA.pas
(*2007 Domestic A*)
{$apptype console} //Delphi

program DomesticA;
var n: integer;

procedure judge(n:integer);
var i: integer;
    ma,mi,x,s: longint;
begin
  ma:=0;mi:=1000;s:=0;
  if(n=0) then exit;
  for i:=1 to n do begin
    readln(x);
    if(x<mi) then mi:=x;
    if(x>ma) then ma:=x;
    s:=s+x;
  end;
  writeln((s-ma-mi) div (n-2));
end;

begin
n:=1;
repeat
  readln(n);
  judge(n);
until(n=0);
end.